sd_meh/merge_methods.py

```diff
@@ -94,12 +96,55 @@ def multiply_difference(
     return c + alpha * difference
 
 
-def transmogrify_distribution(a: Tensor, b: Tensor, alpha: float, **kwargs) -> Tensor:
-    a_dist = torch.msort(torch.flatten(a))
+def top_k_tensor_sum(
+    a: Tensor, b: Tensor, alpha: float, beta: float, **kwargs
+) -> Tensor:
+    a_flat = torch.flatten(a)
+    a_dist = torch.msort(a_flat)
     b_indices = torch.argsort(torch.flatten(b), stable=True)
-    a_redist = torch.gather(a_dist, 0, torch.argsort(b_indices))
-    a_trans = a_redist.reshape(a.shape)
-    return (1 - alpha) * b + alpha * a_trans
+    redist_indices = torch.argsort(b_indices)
+
+    start_i, end_i, region_is_inverted = ratio_to_region(alpha, beta, torch.numel(a))
+    start_top_k = kth_abs_value(a_dist, start_i)
+    end_top_k = kth_abs_value(a_dist, end_i)
+
+    indices_mask = (start_top_k < torch.abs(a_dist)) & (torch.abs(a_dist) <= end_top_k)
+    if region_is_inverted:
+        indices_mask = ~indices_mask
+    indices_mask = torch.gather(indices_mask.float(), 0, redist_indices)
+
+    a_redist = torch.gather(a_dist, 0, redist_indices)
+    a_redist = (1 - indices_mask) * a_flat + indices_mask * a_redist
+    return a_redist.reshape_as(a)
+
+
+def kth_abs_value(a: Tensor, k: int) -> Tensor:
```

Comment:
Can't `torch.abs` the result of kthvalue as it will give a different kth value. Also, black doesn't like me when I try to feed it something as long as that 😅